### PR TITLE
fix: add config for preserving window proportions on open file

### PR DESCRIPTION
**Pull request by rlch**
[Before](https://asciinema.org/a/kUQ5jCy8oujceqTtZFye5tAym)
[After](https://asciinema.org/a/D4jZgcfvyysKBmtqfcquBtadj)

Please let me know if there's anything I haven't considered in this PR

### CODE REVIEW

File: lua/nvim-tree/view.lua [hunk 1]
```diff
@@ -179,7 +179,10 @@ function M.resize(size)
   else
     a.nvim_win_set_height(M.get_winnr(), get_size())
   end
-  vim.cmd ":wincmd ="
+
+	if vim.g.nvim_tree_preserve_window_proportions ~= 1 then
```
Comment (kyazdani42) [inline lua/nvim-tree/view.lua:183]: use a setup option please, i'm removing all vim global options iteratively.

Comment (kyazdani42) [reply lua/nvim-tree/view.lua:183]: something like `view.preserve_window_propertions`

File: doc/nvim-tree-lua.txt [hunk 2]
```diff
@@ -537,7 +543,6 @@ selectable. The default table is
       }
     }
 <
-
```
Comment (kyazdani42) [inline doc/nvim-tree-lua.txt:540]: uneeded change

**Review by kyazdani42 — COMMENTED** [review]
Can you also add the default inside the default mappings in nvim-tree.lua ?
Also, add the default inside the help doc setup :)

**Comment by rlch** [comment]
> Can you also add the default inside the default mappings in nvim-tree.lua ? 

The defaults are also in `view.lua`, is this fine or should they only be defined in one file?

**Comment by kyazdani42** [comment]
Hum i made a mistake sorry, could you remove the view setup you added and add it in the view.lua file directly ? Sorry for the inconvenience

**Review by kyazdani42 — APPROVED** [review]
PR is good now, but i realized when opening a file, it will still equalize the windows because of the resize function.

**Comment by rlch** [comment]
>  but i realized when opening a file, it will still equalize the windows because of the resize function.

What do you mean by this? When I open a file with `preserve_window_proportions=true` it doesn't equalize the windows. Is there an edge-case I haven't considered in the [demo](https://asciinema.org/a/D4jZgcfvyysKBmtqfcquBtadj)?

**Comment by kyazdani42** [comment]
in some case, line 214 open-file will call view.resize which does `wincmd =`

**Comment by rlch** [comment]
> in some case, line 214 open-file will call view.resize which does `wincmd =`

```lua
if not M.View.preserve_window_proportions then
  vim.cmd(":wincmd =")
end
```
Given the `wincmd =` in `view.resize` is guarded by the option, this won't be called form `open-file` right?

**Review by kyazdani42 — APPROVED** [review]
jeez i thought it was another function this was guarded :smile:

**Pull request merged**